cpp/density/LocalDensity.cc

```diff
@@ -12,7 +12,17 @@ namespace freud { namespace density {
 
 LocalDensity::LocalDensity(float r_max, float diameter)
     : m_box(box::Box()), m_r_max(r_max), m_diameter(diameter)
-{}
+{
+    if (r_max <= 0)
+    {
+        throw std::invalid_argument("Local density calculations require that r_max must be positive");
```

Comment:
This suggestion matches the style of the existing error messages more closely. See #825 for other examples.
```suggestion
        throw std::invalid_argument("LocalDensity requires r_max to be positive.");
```